mbbsd/mail.c

```diff
@@ -1244,6 +1272,39 @@ mail_account_sysop(void)
         return DIRCHANGED;
     }
 
+    clear();
+    show_mail_account_sysop_desc();
+    pressanykey();
+
+    clear();
+    vs_hdr("�п�ܭn�H�����@��b������"); 
+    for (i = 0; i < Vector_length(&namelist) && i + 2 < 24; i++) {
+        const char *userid = Vector_get(&namelist, i);
+        snprintf(linebuf, sizeof(linebuf), "%2d. %s", i + 1, userid);
+        move(i + 2, 0);
+        outs(linebuf);
+    }
+    getdata(1, 0, "�п�J�b������s��: ", linebuf, 4, LCECHO);
+    if (linebuf[0] < '1' || linebuf[0] > '9')
+    {
+        vmsg("��J�s�������T");
+        Vector_delete(&namelist);
+        return DIRCHANGED;
+    }
+    sscanf(linebuf, "%d", &sysop_index);
+    if (sysop_index < 1 || sysop_index > Vector_length(&namelist))
+    {
+        vmsg("��J�s�������T");
+        Vector_delete(&namelist);
+        return DIRCHANGED;
+    }
```

Comment:
Could you make these together so the code can be shorter?
if (sscanf(...) != 1 || index < 1 || index > Vector_length(...)) { /* error */ }
The implementation is slightly different (accepts <space>1) but I think the original check (only first char) is not adequate either.